archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericFieldTypesTest.java

```diff
@@ -0,0 +1,522 @@
+package com.tngtech.archunit.core.importer;
+
+import java.io.File;
+import java.io.Serializable;
+import java.lang.ref.Reference;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.domain.JavaType;
+import com.tngtech.archunit.testutil.ArchConfigurationRule;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static com.tngtech.archunit.testutil.Assertions.assertThatType;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteClass.concreteClass;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteParameterizedType.parameterizedType;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteTypeVariable.typeVariable;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteTypeVariableArray.typeVariableArray;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteWildcardType.wildcardType;
+
+@RunWith(DataProviderRunner.class)
+public class ClassFileImporterGenericFieldTypesTest {
+
+    @Rule
+    public final ArchConfigurationRule configurationRule = new ArchConfigurationRule().resolveAdditionalDependenciesFromClassPath(false);
+
+    @Test
+    public void imports_non_generic_field_type() {
+        class NonGenericFieldType {
+        }
+        @SuppressWarnings("unused")
+        class SomeClass {
+            NonGenericFieldType field;
+        }
+
+        JavaType fieldType = new ClassFileImporter().importClass(SomeClass.class).getField("field").getType();
+
+        assertThatType(fieldType).as("generic field type").matches(NonGenericFieldType.class);
+    }
+
+    @Test
+    public void imports_generic_field_type_with_one_type_argument() {
+        @SuppressWarnings("unused")
+        class GenericFieldType<T> {
+        }
+        @SuppressWarnings("unused")
+        class SomeClass {
+            GenericFieldType<String> field;
```

Comment:
If this was `GenericFieldType<String>[] field`, what would `genericFieldType` be like?
I'd have expected that its erasure was `GenericFieldType[].class`, but this doesn't seem to be the case.

----
By the way, for a `NonGenericFieldType[] field` in the test above,
```
 assertThatType(fieldType).as("generic field type").hasErasure(NonGenericFieldType[].class)
```
works, but
```
assertThatType(fieldType).as("generic field type").matches(NonGenericFieldType[].class);
```
doesn't.

Reply:
For the first part: Yes, that might be an error in `toErasure()` :thinking: 
For the second part, this is likely, because `Foo[]` can't be imported from any class file. So maybe there is a discrepancy in the behavior for missing type creation for array types with respect to the Reflection API :thinking: